include/aspect/particle/particle_handler.h

```diff
@@ -144,9 +199,60 @@ namespace aspect
         types::particle_index n_locally_owned_particles() const;
 
         /**
+         * Return a reference to the property pool that owns all particle
+         * properties, and organizes them physically.
+         */
+        PropertyPool &
+        get_property_pool() const;
+
+        /**
          * Return the number of particles in the given cell.
          */
-        unsigned int n_particles_in_cell(const typename Triangulation<dim>::active_cell_iterator &cell) const;
+        unsigned int
+        n_particles_in_cell(const typename Triangulation<dim,spacedim>::active_cell_iterator &cell) const;
+
+        /**
+         * Get a map between subdomain id and the neighbor index. In other words
+         * the returned map answers the question: Given a subdomain id, which
+         * neighbor of the current processor's domain (in terms of a contiguous
+         * number from 0 to n_neighbors) owns this subdomain?
+         */
+        std::map<types::subdomain_id, unsigned int>
+        get_subdomain_id_to_neighbor_map() const;
+
+        /**
+         * Returns a vector that contains a tensor for every vertex-cell
+         * combination of the output of dealii::GridTools::vertex_to_cell_map()
+         * (which is expected as input parameter for this function).
+         * Each tensor represents a geometric vector from the vertex to the
+         * respective cell center.
+         */
+        std::vector<std::vector<Tensor<1,spacedim> > >
+        vertex_to_cell_centers_directions(const std::vector<std::set<typename parallel::distributed::Triangulation<dim,spacedim>::active_cell_iterator> > &vertex_to_cells) const;
+
+        /**
+         * Finds the cells containing each particle for all particles in
+         * @p particles_to_sort. If particles moved out of the local subdomain
+         * they will be sent to their new process and inserted there.
+         * After this function call every particle is either on its current
+         * process and in its current cell, or deleted (if it could not find
+         * its new process or cell).
+         *
+         * @param [in] particles_to_sort Vector containing all pairs of
+         * particles and their old cells that will be sorted into the
+         * 'particles' member variable in this function.
+         */
+        void
+        sort_particles_in_subdomains_and_cells();
```

Comment:
the documentation here does not match the signature any more, as it talks about `particles_to_sort`.